aws_s3_bucket.tf

```diff
@@ -1,41 +1,95 @@
 resource "aws_s3_bucket" "bucket" {
   bucket        = var.s3_bucket_name
-  policy        = var.s3_bucket_policy
-  acl           = var.s3_bucket_acl
   force_destroy = var.s3_bucket_force_destroy
 
-  dynamic "versioning" {
-    for_each = length(keys(var.versioning)) == 0 ? [] : [var.versioning]
+  tags = merge(var.common_tags)
+}
 
-    content {
-      enabled    = lookup(versioning.value, "enabled", null)
-      mfa_delete = lookup(versioning.value, "mfa_delete", null)
-    }
+resource "aws_s3_bucket_public_access_block" "bucket_access" {
+  bucket = aws_s3_bucket.bucket.id
+
+  block_public_acls       = var.block_public_acls
+  block_public_policy     = var.block_public_policy
+  ignore_public_acls      = var.ignore_public_acls
+  restrict_public_buckets = var.restrict_public_buckets
+}
+
+resource "aws_s3_bucket_acl" "bucket-acl" {
+  count = var.s3_bucket_acl == null || var.s3_bucket_acl == "" || var.s3_bucket_acl == "private" ? 1 : 0
```

Comment:
If any of these are `null`, `""` or `private` then bucket_acl resource is created?

Reply:
Yes indeed. This was more to cater for existing TF folders having those values.